src/Symfony/Component/HttpFoundation/IpUtils.php

```diff
@@ -61,6 +66,60 @@ public static function checkIp($requestIp, $ips)
      */
     public static function checkIp4($requestIp, $ip)
     {
+        return self::checkIpInternal($requestIp, $ip, 'self::checkIp4Internal');
```

Comment:
I would not delegate to another method but just copy/paste the cache logic in the methods directly.